testng-core/src/main/java/org/testng/internal/invokers/MethodInvocationHelper.java

```diff
@@ -417,6 +423,18 @@ private static void invokeWithTimeoutWithNewExecutor(
     }
   }
 
+  private static StackTraceElement[] getRunningMethodStackTrace(ExecutorService exec) {
+    if (!(exec instanceof ThreadPoolExecutor)) return null;
+    ThreadFactory threadFactory = ((ThreadPoolExecutor) exec).getThreadFactory();
+
+    if (!(threadFactory instanceof TestNGThreadFactory)) return null;
+    TestNGThreadFactory testNGThreadFactory = (TestNGThreadFactory) threadFactory;
+
+    Collection<Thread> threads = testNGThreadFactory.getRunningThreads();
+    if (threads.size() == 1) return threads.iterator().next().getStackTrace();
+    return null;
```

Comment:
Any way to get the stacktrace if there is more than 1 thread?

Reply:
Currently, only one thread is submitted in the newly created executor, so there can only be one thread.
I think this discussion should be made in a future issue that will add multiple threads.

Note that `ITestResult` supports only one `Throwable` with one stacktrace.

Reply:
I propose something like:

```
switch (threads.size()) {
  case 0: return null;
  case 1: return threads.iterator().next().getStackTrace();
  default: "log warning" and return null or the first stackTrace
```

It will prevent strange behaviors if the executor has more than one thread in the future.